### ISSUE TITLE
Kind of too complex, no way to setup routing right inside the page

### ISSUE THREAD

**Issue by hrumhurum** [aspnet/Docs#7107]
This page is very hard to follow as it combines several different concepts into single chapter which is a whooping 24 pages long. Sorry, that looks needlessly complicated. Cannot make any use of it without significant pains.

In contrast, [Route] attribute from ASP.NET MVC is just one line away from beautiful structured URLs. Compare that one-liner to these multiline Frankensteins.

What I want is to setup a route *inside* the page (or page's controller), not in that UseXXX.Options.Options.Options god object. Is there a way to do so?

P.S.: Just found a way to specify the route right inside the page. Like this: 

`@page "yorfolder123\yourpage123"`

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ef2dfc62-6ac3-101b-88cb-9f5e7312d311
* Version Independent ID: 08dd4682-51f4-ca04-a6f4-8d49403b609f
* Content: [Razor Pages route and app conventions in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/razor-pages/razor-pages-conventions?view=aspnetcore-2.1)
* Content Source: [aspnetcore/mvc/razor-pages/razor-pages-conventions.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/mvc/razor-pages/razor-pages-conventions.md)
* Product: **asp.net-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
> P.S.: Just found a way to specify the route right inside the page. Like this:
> @page "yorfolder123\yourpage123"

Yes, that's correct. I think cross-linking to the basic page routing would have helped in this case. The topic is meant to explore the more advanced conventions.

I have to run out now for a couple of hours. I'll provide a more detailed response when I get back. Let's use this issue to work on the cross-linking. Thanks @hrumhurum for commenting.

**Comment by guardrex** [comment]
Back! :smile:

Yes, I'm going to do something now to give readers who land on this topic a chance to "run for the border" early in the coverage to get over to our content on `@page` directive routing.

The only possible split here would be for the last section, *Replace the default page app model provider*, and even then I don't think it would help all that much overall.

**Comment by guardrex** [comment]
I addressed the problem on #7123.

I think most of your concerns stemmed from looking for how custom routes work with Razor Pages not because this topic is faulty. The conventions described in this topic are correct, and we'd like to keep them together in one spot ... at least for the time being. If engineering comes up with *more* convention scenarios, then 'yes,' I think we'd want to consider breaking this up.

On #7123, I seek to solve the problem for a reader landing here by linking out early over to a revised "Custom routes" section of the introduction to RP topic. That should do the trick. If you think that still wouldn't have helped you, let me know.

**Comment by hrumhurum** [comment]
@guardrex  I appreciate your reactivity. Changes in #7123 do the job. Thanks a lot.
